parser/prism/Translator.cc

```diff
@@ -939,9 +939,23 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
             auto name = translate(moduleNode->constant_path);
             auto declLoc = translateLoc(moduleNode->module_keyword_loc).join(name->loc);
 
-            auto body = translate(moduleNode->body);
+            auto moduleTranslator = enterModule();
+            auto bodyNode = moduleTranslator.translate(moduleNode->body);
 
-            return make_unique<parser::Module>(location, declLoc, move(name), move(body));
+            if (!hasExpr(name)) {
+                return make_unique<parser::Module>(location, declLoc, move(name), move(bodyNode));
+            }
+
+            auto bodyStore = bodyToRHSStore(bodyNode);
+            if (!bodyStore.has_value()) {
+                return make_unique<parser::Module>(location, declLoc, move(name), move(bodyNode));
+            }
+
+            auto nameExpr = name->takeDesugaredExpr();
+            ast::ClassDef::ANCESTORS_store ancestors;
+            auto expr = MK::Module(location, declLoc, std::move(nameExpr), std::move(ancestors), std::move(bodyStore.value()));
```

Comment:
[`std::optional.value()`](https://en.cppreference.com/w/cpp/utility/optional/value.html) does a checked access to throw `std::bad_optional_access` in case it's a `nullopt`. We know that can't be the case here because of the `has_value()` check above, so you can just do an unchecked access via [`operator*`](https://en.cppreference.com/w/cpp/utility/optional/operator%252A.html)


```suggestion
            auto expr = MK::Module(location, declLoc, move(nameExpr), move(ancestors), move(*bodyStore));
```